packages/react-dev-utils/checkRequiredFiles.js

```diff
@@ -7,24 +7,51 @@
 
 'use strict';
 
-var fs = require('fs');
-var path = require('path');
-var chalk = require('chalk');
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+
+// Currently, files passed without an extension are assumed to be js-like.
+// If other types of files need to be passed later without extensions the
+// checkRequiredFiles function will need to be modified.
+const fallbackExts = [
+  '.web.js',
+  '.mjs',
+  '.js',
+  '.web.jsx',
+  '.jsx',
+  '.ts',
+  '.tsx',
+];
+const withExt = (ext, file) => file.dir + path.sep + file.name + ext;
 
 function checkRequiredFiles(files) {
-  var currentFilePath;
+  let didFallbackExts = false;
   try {
-    files.forEach(filePath => {
-      currentFilePath = filePath;
-      fs.accessSync(filePath, fs.F_OK);
-    });
+    for (const filePath of files) {
+      const file = path.parse(filePath);
+
+      if (file.ext === '') {
+        didFallbackExts = true;
+        if (fallbackExts.some(e => fs.existsSync(withExt(e, file)))) {
+          continue;
+        } else {
+          throw file;
+        }
+      }
+
+      if (!fs.existsSync(filePath)) throw file;
+    }
     return true;
-  } catch (err) {
-    var dirName = path.dirname(currentFilePath);
-    var fileName = path.basename(currentFilePath);
+  } catch (file) {
     console.log(chalk.red('Could not find a required file.'));
-    console.log(chalk.red('  Name: ') + chalk.cyan(fileName));
-    console.log(chalk.red('  Searched in: ') + chalk.cyan(dirName));
+    console.log(chalk.red('  Name: ') + chalk.cyan(file.base));
+    if (didFallbackExts) {
+      // Typescript files will pass the check but are not officially supported.
+      const extStr = fallbackExts.filter(e => /\.(?!tsx?)/.test(e)).join(', ');
```

Comment:
I included `.ts` and `.tsx` files but filtered them from the "Supported extensions" string. If you're not ok with this I'll change it, but I'd really love to see (and help with) any movement towards supporting Typescript, even if it's unofficial/undocumented just to make life easier for packages like [the one I'm using](https://github.com/strothj/react-app-rewire-typescript-babel-preset). I realize the README suggests using  `create-react-app-typescript`, but I prefer `react-app-rewired` because it works more seamlessly and doesn't "lag behind" when you guys update, and I love to stay on the bleeding edge!